File: packages/schemas/src/item.ts
import { z } from "zod";

/**
 * Item — the canonical example entity from the golden-stack spec.
 * The matching Pydantic model lives in `python/platform_schemas/item.py`.
 * Keep the two shapes in lockstep: this is the single source of truth.
 */
export const Item = z.object({
  id: z.string().uuid(),
  name: z.string().min(1),
  created_at: z.string().datetime(),
});
export type Item = z.infer<typeof Item>;

export const ItemCreate = Item.pick({ name: true });
export type ItemCreate = z.infer<typeof ItemCreate>;

File: packages/schemas/python/platform_schemas/item.py
"""Item — Pydantic v2 mirror of the zod schema in ../../src/item.ts.

Single source of truth: the two shapes (zod + Pydantic) must stay in lockstep.
Pydantic v2 only (model_config / field_validator) — never v1.
"""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class Item(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    created_at: datetime


class ItemCreate(BaseModel):
    name: str
